service/instance.py

```diff
@@ -1521,13 +1522,25 @@ def _to_user_driver(core_identity):
         (auth, sess, token) = _token_to_keystone_scoped_project(
             auth_url, auth_token,
             project_name, domain_name)
+        user_driver = UserManager(auth_url=auth_url,
```

Comment:
I'm not 100% clear on how this stuff works under the hood, but why do you pass session, auth_token, and auth_url?  I thought session included the other 2 somehow.  Or is it just covering the bases in case some of the above are not there?